Jenkinsfile

```diff
@@ -0,0 +1,247 @@
+        stage('Shell tests') {
+            executeShellTests()
+        }
+
+        stage('Generate k8s Resources') {
+            docker.image("golang:${goVersion}")
+                    .mountJenkinsUser()
+                    .inside("--volume ${WORKSPACE}:/workdir -w /workdir") {
+                        make 'k8s-create-temporary-resource'
+                    }
+            archiveArtifacts 'target/make/k8s/*.yaml'
+        }
+
+        K3d k3d = new K3d(this, "${WORKSPACE}", "${WORKSPACE}/k3d", env.PATH)
+
+        try {
+            String doguVersion = getDoguVersion(false)
+            GString sourceDeploymentYaml = "target/make/k8s/${repositoryName}_${doguVersion}.yaml"
+
+            stage('Set up k3d cluster') {
+                k3d.startK3d()
+            }
+
+            String imageName
+            stage('Build & Push Image') {
+                // pull protected base image
+                docker.withRegistry('https://registry.cloudogu.com/', "cesmarvin-setup") {
+                    String currentBaseImage = sh(
+                        script: 'grep -m1 "registry.cloudogu.com/official/base" Dockerfile | sed "s|FROM ||g" | sed "s| as builder||g"',
+                        returnStdout: true
+                    )
+                    currentBaseImage = currentBaseImage.trim()
+                    image = docker.image(currentBaseImage)
+                    image.pull()
+                }
+
+                String namespace = getDoguNamespace()
+                imageName = k3d.buildAndPushToLocalRegistry("${namespace}/${repositoryName}", doguVersion)
+            }
+
+            stage('Setup') {
+                k3d.setup("v0.6.0", [
+                        dependencies: ["official/postfix", "official/plantuml", "k8s/nginx-ingress"],
+                        defaultDogu : "plantuml"
+                ])
+            }
+
+            stage('Deploy Dogu') {
+                k3d.installDogu(repositoryName, imageName, sourceDeploymentYaml)
+            }
+
+            stage('Wait for Ready Rollout') {
+                k3d.waitForDeploymentRollout(repositoryName, 300, 5)
+            }
+
+            stage('Test static content') {
+                testStaticContentAccess(k3d)
+            }
+
+            stageAutomaticRelease()
+        } finally {
+            stage('Remove k3d cluster') {
+                k3d.deleteK3d()
+            }
+        }
+    }
+}
+
+
+/**
+ * Checks whether the static content is accessible and also the correct content.
+ */
+void testStaticContentAccess(K3d k3d) {
+    k3d.waitForDeploymentRollout("nginx-ingress", 300, 5)
+
+    // determine free port with python
+    String port = sh(script: 'echo -n $(python3 -c \'import socket; s=socket.socket(); s.bind(("", 0)); print(s.getsockname()[1]); s.close()\');', returnStdout: true)
+
+    // the process is automatically terminated when canceling/terminating the build
+    k3d.kubectl("port-forward service/nginx-ingress ${port}:443 &")
+
+    sh "sleep 5"
```

Comment:
It is possibile to wait here for the port-forward with a timeout?

Reply:
The execution of line 130 automatically waits for the port forwarding. I do not know anymore why the sleep is needed at all.

I will try to remove the sleep.